### ISSUE TITLE
Print dependency package name when Algolia throws error

### ISSUE THREAD

**Issue by karlhorky** [upleveled/preflight#411]
Preflight currently outputs some confusing error messages like `ObjectID does not exist` from Algolia:

```
⚠ No dependency problems
  ✓ No unused dependencies
  ✖ No dependencies without types
    › ObjectID does not exist
```

Would be great to see what package is causing this in the output, eg. something like this:

```
⚠ No dependency problems
  ✓ No unused dependencies
  ✖ No dependencies without types
    › Algolia error for `packagename`: ObjectID does not exist
```

Maybe a `try`/`catch` around the `await index.getObjectAlgoliaObj`, rethrowing the error in the `catch` block and adding `dependency` to the string?

https://github.com/upleveled/preflight/blob/19ce5fb9faf30c447999110b9e069e881918ccac/src/checks/noDependencyProblems/noDependenciesWithoutTypes.ts#L33-L69

**Comment by karlhorky** [comment]
Closed by #412

**Comment by Eprince-hub** [comment]
Algolia is failing for the `smart-image-saver` package
```js
$  preflight
🚀 UpLeveled Preflight v2.1.16
✔ All changes committed to Git
✔ node_modules/ folder ignored in Git
✔ No extraneous files committed to Git
✔ No secrets committed to Git
✔ Use single package manager
✔ Project folder name matches correct format
⚠ No dependency problems
  ✔ No unused dependencies
  ✖ No dependencies without types
    › Algolia error for `smart-image-saver`: ObjectID does not exist
✔ GitHub repo has deployed project link under About
✖ ESLint
  › ESLint problems found in the following files:
  index.js

  Open these files in your editor - there should be problems to fix
✔ Prettier
✔ ESLint config is latest version
✔ Preflight is latest version
```
